### PR TITLE
Modified server structure and added login backend to communicate with MongoDB

### PR DESCRIPTION
**Pull request by Lord-Bubbles**
#12 #21

### CODE REVIEW

File: src/controllers/UserController.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../database/UserModel");
+
+// @desc   Adds new user if it doesnt exist and returns newly created user, otherwise return existing user from database
```
Comment (chilimalone) [inline src/controllers/UserController.js:4]: nit - you want a tiny bit more info here. what defines a unique user? by username? by email? etc

File: src/controllers/UserController.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    res.status(200).json({ 
+      data: user, 
+      message: "User already exists. Returning user information" 
+    });
+  } else {
+    user = await User.create({ email: email, name: name, recipes: [] });
+    if (user) {
+      res.status(200).json({
+        data: {
+          _id: user._id,
```
Comment (chilimalone) [inline src/controllers/UserController.js:21]: this organization is not ideal. You are returning all the user's recipes when you just want to get the user's info (presumably an identifier that allows you to retrieve more information about the user for later).

right now you only have a recipes page, but in the future if/when you have other things, you'll be passing and processing a lot of information across the wire that are not needed for the page to function (e.g. sending recipes info for a schedule/calendar page)

File: src/routes/RecipeRoutes.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const express = require("express");
+const router = express.Router();
+
+module.exports = router;
```
Comment (chilimalone) [inline src/routes/RecipeRoutes.js:4]: nit: newline at the end of the file.

File: src/routes/UserRoutes.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const router = express.Router();
+const {addOrUpdateUser} = require("../controllers/UserController")
+
+// @desc   Adds new user if it doesnt exist and returns newly created user, otherwise return existing user from database
+// @route  PUT /users
+// @access Public
+router.put("/", addOrUpdateUser);
+
+module.exports = router;
```
Comment (chilimalone) [inline src/routes/UserRoutes.js:10]: ditto new line at end of file

File: src/controllers/UserController.js (hunk 2 above)
Comment (Lord-Bubbles) [reply src/controllers/UserController.js:21]: acknowledged and modified

File: src/routes/RecipeRoutes.js (hunk 3 above)
Comment (Lord-Bubbles) [reply src/routes/RecipeRoutes.js:4]: done

File: src/routes/UserRoutes.js (hunk 4 above)
Comment (Lord-Bubbles) [reply src/routes/UserRoutes.js:10]: done

File: src/controllers/UserController.js (hunk 1 above)
Comment (Lord-Bubbles) [reply src/controllers/UserController.js:4]: added description defining a unique user

File: src/database/Config.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    const conn = await mongoose.connect(process.env.MONGODB_URI);
+    
+    console.log(`MongoDB Connected: ${conn.connection.host}`);
+  } catch (error) {
+    console.log(error);
+    process.exit(1);
+  }
+}
+
+module.exports = connectDB;
```
Comment (chilimalone) [inline src/database/Config.js:15]: nit: newline at end of page

**Review by chilimalone — APPROVED** [review]

File: src/database/Config.js (hunk 5 above)
Comment (Lord-Bubbles) [reply src/database/Config.js]: done

**Pull request merged**